index.ts

```diff
@@ -113,32 +130,33 @@ function candidateToGeojson (candidate) {
  * @param {string} [$0.url]                       optional URL to override ESRI suggest endpoint
  * @return {Promise}                              A Promise that'll get resolved with the suggest result
  */
-export function autocomplete ({
+export function autocomplete({
+  boundary,
   clientId,
   clientSecret,
-  boundary,
   focusPoint,
   text,
   url
 }: BaseQuery & {
-  boundary?: Boundary,
-  focusPoint?: any,
-  text: string,
-}): Promise<BaseResponse> {
-  const geocoder = getGeocoder(clientId, clientSecret, url)
-  const options = {}
+  boundary?: Boundary
+  focusPoint?: Point
+  text?: string
+}): Promise<GeoJSON.FeatureCollection & { query: { text: string } }> {
```

Comment:
This does not actually return GeoJSON. It returns the data nearly verbatim from the [suggest](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-suggest.htm) API endpoint. I would write this type as follows:

```suggestion
}): Promise<{
  features: Array<SuggestOption>
  query: { text: string }
}> {
```

where `SuggestOption` is defined as:

```js
type SuggestOption = {
  isCollection: boolean
  magicKey: string
  text: string
}
```

Reply:
Refactored in aecf758d679984328a465df451fd953bde17efbf